ledger/src/blockstore_db.rs

```diff
@@ -254,6 +254,18 @@ struct Rocks {
     write_batch_perf_status: PerfSamplingStatus,
 }
 
+macro_rules! rocks_try {
+    ($expression:expr) => {{
+        match $expression {
+            Ok(output) => output,
+            Err(error) => {
+                $crate::blockstore_db::Rocks::print_error_handling_instruction(&error);
+                return Err($crate::blockstore_db::BlockstoreError::RocksDb(error));
+            }
+        }
+    }};
+}
+
 impl Rocks {
     fn open(path: &Path, options: BlockstoreOptions) -> Result<Rocks> {
```

Comment:
according to past bug reports, wrapping `open` isn't enough, i guess? these corruption error needs a full scan of entire sst files. i doubt it will be done everytime when opening db. ;)

Reply:
RocksDB reports corruption only when it actually read that block, which IMO is a good thing because this will allow us to open the db and obtain file metadata in our ledger tool.  For instance, #26790 allows our ledger tool to print out slot range of a sst file, and #26905 allows our ledger tool to purge older slots to free up space.  They can both operate on corrupted sst files.